templates/linux_user_data.tpl

```diff
@@ -1,12 +1,48 @@
 %{ if enable_bootstrap_user_data ~}
+MIME-Version: 1.0
+Content-Type: multipart/mixed; boundary="==========20221208145300=="
+
+--==========20221208145300==
+Content-Type: text/x-shellscript; charset="us-ascii"
+MIME-Version: 1.0
+Content-Transfer-Encoding: 7bit
+Content-Disposition: attachment
+
 #!/bin/bash
-set -e
+set -ex
+
 %{ endif ~}
+%{ if length(pre_bootstrap_user_data) > 0 ~}
+# User-supplied pre userdata
 ${pre_bootstrap_user_data ~}
+%{ endif ~}
+%{ if format_mount_nvme_disk ~}
+echo "Format and Mount NVMe Disks if available"
+IDX=1
+DEVICES=$(lsblk -o NAME,TYPE -dsn | awk '/disk/ {print $1}')
+
+for DEV in $DEVICES
+do
+  mkfs.xfs /dev/$${DEV}
+  mkdir -p /local$${IDX}
+
+  echo /dev/$${DEV} /local$${IDX} xfs defaults,noatime 1 2 >> /etc/fstab
+
+  IDX=$(($${IDX} + 1))
+done
+mount -a
+%{ endif ~}
+%{ if length(cluster_service_ipv4_cidr) > 0 ~}
+export SERVICE_IPV4_CIDR=${cluster_service_ipv4_cidr}
+%{ endif ~}
 %{ if enable_bootstrap_user_data ~}
 B64_CLUSTER_CA=${cluster_auth_base64}
 API_SERVER_URL=${cluster_endpoint}
-/etc/eks/bootstrap.sh ${cluster_name} ${bootstrap_extra_args} --b64-cluster-ca $B64_CLUSTER_CA --apiserver-endpoint $API_SERVER_URL \
-  --ip-family ${cluster_ip_family} --service-${cluster_ip_family}-cidr ${cluster_service_cidr}
```

Comment:
Are we sure we want to skip these additions to the bootstrap.sh script?

Reply:
They did not exist on the old fork, I think we should be keeping these last two arguments: https://github.com/pinterest/terraform-aws-eks/pull/3/files#diff-5e40a572eaa3c9cddd3c65bbbff6fc1477976db1b37e3b56e4d2aaa914b94eabL12

Reply:
I take it back, looks like both of these are inferred from the describe cluster query so our arguments should be fine: https://github.com/awslabs/amazon-eks-ami/blob/2b5acb97f4c6d1971cb7d645e1fdb13b92cee10e/templates/al2/runtime/bootstrap.sh#L387-L391